src/experiment/InterpolateExperiment.py

```diff
@@ -1,51 +1,99 @@
-
-            interpolated = self.__interpolate.interpolateAll(interpolateSubdataset.xLeft, interpolateSubdataset.xRight, 2)
-            interpolateSubdatasetArrays = [
-                interpolateSubdataset.xLeft,
-                self.__autoencoder.autoencoder().predict_on_batch(interpolateSubdataset.xLeft),
-                interpolateSubdataset.xRight,
-                self.__autoencoder.autoencoder().predict_on_batch(interpolateSubdataset.xRight),
-                interpolated[:, 2]
-            ]
-            if interpolateSubdataset.centreIsSpecified():
-                interpolateSubdatasetArrays.append(interpolateSubdataset.xCentre)
-                interpolateSubdatasetArrays.append(self.__autoencoder.autoencoder().predict_on_batch(interpolateSubdataset.xCentre))
-                if interpolateSubdataset.outsideIsSpecified():
-                    interpolateSubdatasetArrays.append(interpolateSubdataset.xOutside)
-                    interpolateSubdatasetArrays.append(self.__autoencoder.autoencoder().predict_on_batch(interpolateSubdataset.xOutside))
-            imagesArrayComparisonDisplay(interpolateSubdatasetArrays, self.__resultRouteStem + "interpolate_subdataset_" + interpolateSubdataset.interpolatedFactorName + ".png")
+        for interpolationSubdataset in self.__interpolateData:
+            self.__numericalInterpolationMetrics(interpolationSubdataset)
+            self.__visualInterpolation(interpolationSubdataset)
+
+        if len(self.__interpolateData) > 1:
+            combineSubdatasets = CombineInterpolateLoaders()
+            combinedSubdatasets = combineSubdatasets.combine(self.__interpolateData, "COMBINED")
+            self.__numericalInterpolationMetrics(combinedSubdatasets)
+            self.__visualInterpolation(combinedSubdatasets)
+
+    def __numericalInterpolationMetrics(self, interpolateSubdataset: InterpolateSubdataset):
+        interpolated, interpolatedReconstructed = self.__interpolateLatentSpace.interpolateAll(interpolateSubdataset.xLeft, interpolateSubdataset.xRight, 2)
+
+        interpolatedReconstructed = interpolatedReconstructed[:, 1]
+        interpolated = interpolated[:, 1]
+
+        if interpolateSubdataset.centreIsSpecified():
+            self.__resultsStore.storeValue(
+                [self.__datasetName, self.__modelName, interpolateSubdataset.interpolatedFactorName, "actual", "metricImageSpace"],
+                self.__imageSpaceComparisonMetric.compute(interpolateSubdataset.xCentre, interpolatedReconstructed).mean
+            )
+
+            self.__resultsStore.storeValue(
+                [self.__datasetName, self.__modelName, interpolateSubdataset.interpolatedFactorName, "actual", "metricLatentSpace"],
+                self.__latentSpaceComparisonMetric.compute(
+                    self.__autoencoder.encoder().predict(interpolateSubdataset.xCentre, batch_size=100),
+                    interpolated
+                ).mean
+            )
```

Comment:
Add interpolation in the image space as a point of comparison, and also random picture